src/lib.rs

```diff
@@ -228,7 +228,7 @@ might be easier to debug than a deadlock.
 #[path = "imp_pl.rs"]
 mod imp;
 
-#[cfg(feature = "std")]
+#[cfg(any(feature = "std", feature = "assume-no-threads-or-interrupts"))]
```

Comment:
It would be nice if we could just declare the `imp` module inside the `sync` module, so we don't have to worry about these `cfg`s getting out of sync.

However, https://github.com/rust-lang/rust/issues/35016 is still not fixed, so the only way to do this is moving the `imp_*.rs` files (which seems like a hassle).